client/views/2-DashboardView.js

```diff
@@ -0,0 +1,126 @@
+import React, { Component } from 'react';
+import { View, StyleSheet, Button, Text } from 'react-native';
+import { connect } from 'react-redux';
+import { storeContacts, storeMyId, storeMyProfile }  from '../redux/actions/actions';
+
+import HeadersActions from './../components/HeadersActions';
+import ContactsScrollPanel from './../components/ContactsScrollPanel';
+import DashboardActions from './../components/buttons/DashboardActions';
+
+import { backgroundColor, headerColor } from '../utils/style';
+
+const BASE_URL = "http://192.168.1.149:3000/user-friends";
+
+class Dashboard extends Component {
+  state = {
+    noContactsToShow: false
+  }
+  
+  static navigationOptions = {
+    headerLeft: null,
+    headerBackgroundTransitionPreset: 'toggle',
+    title: 'Dashboard',
+    headerRight: (
+      <HeadersActions></HeadersActions>),
+    headerStyle: { backgroundColor: headerColor },
+    headerTintColor: '#fff',
+    headerTitleStyle: {
+      fontWeight: 'bold',
+    },
+  };
+
+  getMyContacts = async () => {
+    console.log('Dashboard - fetch,  this.props.myID',  this.props.myID);
+    await fetch(`${BASE_URL}/${this.props.myID}`, {
+      method: "GET",
+      headers: { 'Content-Type': 'application/json' }
+    })
+    .then(rawData => rawData.json())
+    .then( (contacts) => {
+      // splice last item in the contacts array, as USERS actuall profile was pushed within `getUsersFriends` as last to avoid making double request
+      let myProfile = contacts.splice(contacts.length -1, 1)
+      this.props.storeMyProfile(myProfile[0]);
+      this.props.storeContacts(contacts);
+    })
+    .catch( (err) => {
+      setTimeout( () => {
+        this.setState({noContactsToShow: true});
+        console.log('getMyContactsError',err);
+      }, 500);
+    } 
+    );
+  };
+  
+  componentDidMount () {
```

Comment:
why are you setting a 'setTimeout'? if I think that as the data gets bigger you will have to hard code and extend the list, a better Idea would be to conditionally render here.

let isLoading = this.getMyContacts;

then...

if (isLoading === undefined) {
render(){
  return(
    <Spinner>
    )
} else{
//rest of code for rendering the page
}